nntrainer/layers/layer_context.h

```diff
@@ -51,7 +51,38 @@ class InitLayerContext {
   InitLayerContext(const std::vector<TensorDim> &dim,
                    const std::vector<bool> &req_out_connected, bool in_place_,
                    const std::string &n = "", const std::string &prefix_ = "",
-                   const float max_norm = 0.0);
+                   const float max_norm = 0.0,
+                   std::array<const std::string, 3> tensor_type_ = {
+                     "NCHW", "FP32", "FP32"});
```

Comment:
Here, 
[0] - Format,
[1] - Weight Precision
[2] - Activation Precision

Though no issues with functionality, Isn't it good to follow similar nomenclature of other opensource engines.
Like a separate datatype,
AF32WF32 - Activation Float32, Weight Float 32
AF16WF16
A16W16 etc...

Reply:
May be we can attempt in next PR

Reply:
@jijoongmoon please let me know whether should I be aware of this point during my work for enabling fp16 for every layer in nntrainer